src/URI.js

```diff
@@ -1031,6 +1037,22 @@
     }
   };
 
+  URI.ensureValidPort = function(v) {
+    if (v.match(/[^0-9]/)) {
+      throw new TypeError('Port "' + v + '" contains characters other than [0-9]');
+    }
+  };
+
+  URI.basicValidation = function(parts) {
```

Comment:
I put it here because I just wanted to prevent "http:///" being a valid input for the constructor. Since I am not only checking the hostname, but also protocol and path I extracted it into it's own function.